File: 7/solution.go
package main

import "fmt"

const Len = int(1e6)
var prime = [Len]bool{}

func main() {
	cnt:=0
	for i := 2;i<Len;i++ {
		if !prime[i] {
			cnt ++
			if cnt == 10001 {
				fmt.Printf("%d\n", i)
			}
			for j:=i<<1;j<Len;j+=i {
				prime[j] = true
			}
		}
	}
}

File: 7/solution.cpp
#include <iostream>
using namespace std;
const int len = 1e6;
bool prime[len];
int main() {
    int cnt = 0;
    for (int i = 2; i < len; i++) {
        if (!prime[i]) {
            cnt++;
            if (cnt == 10001)
                cout << i << endl;
            for (int j = i << 1; j < len; j += i)
                prime[j] = true;
        }
    }
    return 0;
}
